library/network_state.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: network_state
+version_added: "2.9"
+short_description: module for network role to apply network state configuration
+description:
+    - This module allows to apply the network state configuration through nmstate,
+      https://github.com/nmstate/nmstate
+options:
+    state:
```

Comment:
what do you think about calling this `current_state`? So it is clear it is the result of the changes..